adagrams/game.py

```diff
@@ -1,11 +1,126 @@
+import random
+
 def draw_letters():
-    pass
+    # Import random - assign the randomized number to the index
+    # "bag" of letters will be a dictionary as shown below
+    # hand will be a list
+# | A : 9  | N : 6 |
+# | B : 2  | O : 8 |
+# | C : 2  | P : 2 |
+# | D : 4  | Q : 1 |
+# | E : 12 | R : 6 |
+# | F : 2  | S : 4 |
+# | G : 3  | T : 6 |
+# | H : 2  | U : 4 |
+# | I : 9  | V : 2 |
+# | J : 1  | W : 2 |
+# | K : 1  | X : 1 |
+# | L : 4  | Y : 2 |
+# | M : 2  | Z : 1 |
+    bag_of_letters_dict = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+    bag_of_letters_list = []
+    for key, value in bag_of_letters_dict.items():
+        for letter in range(value):
+            bag_of_letters_list.append(key)
+
+    hand = []
+
+    while len(hand) < 10:
+        for letter in bag_of_letters_list:
+            hand.append(letter)
+            letter_of_index = bag_of_letters_list.index(letter)
+            bag_of_letters_list.pop(letter_of_index)
+            break
+    return hand
 
 def uses_available_letters(word, letter_bank):
-    pass
+    true_false = []
+    word = word.upper()
+    word_list = []
+
+    for letter in word:
+        word_list.append(letter)
+        if letter in letter_bank and word_list.count(letter) == letter_bank.count(letter):
```

Comment:
This passes our tests, but there is a bug here.  If `letter_bank` has multiple copies of the same letter, even if all of those letters are used in the word, the first time this loop tests one of those letters, this conditional will be False, because the count in word_list will be less than the count in letter_bank.  Consider this examples:
* word = "dog" and letter_bank =  ['d', 'o', 'o', 'g', 'x', 'x', 'x', 'x', 'x', 'x'].  The second iteration through this loop, 'o' will be added to `word_list`, but there will only be one 'o' in word list, so the count here will not be equal.
Changing the conditional to <= would fix this issue, because then it would be checking if there are enough letters in letter_bank, rather than the exact same number.
```suggestion
        if letter in letter_bank and word_list.count(letter) <= letter_bank.count(letter):
```